django/core/paginator.py

```diff
@@ -28,13 +28,30 @@ class Paginator:
     # Translators: String used to replace omitted page numbers in elided page
     # range generated by paginators, e.g. [1, 2, '…', 5, 6, 7, '…', 9, 10].
     ELLIPSIS = _("…")
-
-    def __init__(self, object_list, per_page, orphans=0, allow_empty_first_page=True):
+    default_error_messages = {
+        "page_not_an_integer": _("That page number is not an integer"),
+        "page_not_a_positive_integer": _("That page number is less than 1"),
+        "no_results": _("That page contains no results"),
+    }
+
+    def __init__(
+        self,
+        object_list,
+        per_page,
+        orphans=0,
+        allow_empty_first_page=True,
+        error_messages=None,
+    ):
         self.object_list = object_list
         self._check_object_list_is_ordered()
         self.per_page = int(per_page)
         self.orphans = int(orphans)
         self.allow_empty_first_page = allow_empty_first_page
+        self.error_messages = (
+            self.default_error_messages
+            if error_messages is None
+            else self.default_error_messages | error_messages
```

Comment:
I just learnt `dict_1 | dict_2` merges two dicts. I can use `.copy` and `.update` if it's more friendly|readeable